src/main/java/com/github/hcsp/maven/Version.java

```diff
@@ -13,5 +17,35 @@
      * @param version2 传入的版本字符串2，支持x/x.y/x.y.z，你可以假定传入的字符串一定是合法的语义化版本
      * @return -1/0/1 当version1 小于/等于/大于 version2时
      */
-    public static int compare(String version1, String version2) {}
+    public static int compare(String version1, String version2) {
+        List<String> version1AsList = Arrays.asList(version1.split("\\."));
+        ArrayList<String> newVersion1AsList = new ArrayList<>(version1AsList);
+        List<String> version2AsList = Arrays.asList(version2.split("\\."));
+        ArrayList<String> newVersion2AsList = new ArrayList(version2AsList);
+        int largeVersionNumber = Math.max(newVersion1AsList.size(), newVersion2AsList.size());
+        if (newVersion1AsList.size() < largeVersionNumber) {
+            formatVersion(newVersion1AsList, largeVersionNumber);
+        } else formatVersion(newVersion2AsList, largeVersionNumber);
+
+        for (int i = 0; i < largeVersionNumber; i++) {
+            if (Integer.parseInt(newVersion1AsList.get(i)) > Integer.parseInt(newVersion2AsList.get(i))) {
+                return 1;
+            } else if (Integer.parseInt(newVersion1AsList.get(i)) < Integer.parseInt(newVersion2AsList.get(i))) {
+                return -1;
+            }
+        }
+        return 0;
+    }
+
+    private static void formatVersion(List<String> version, int Number) {
```

Comment:
```suggestion
    private static void formatVersion(List<String> version, int number) {
```

变量的约定都是小写开头，大写会被人误认为类名的。